src/main/java/com/example/WeatherDataCodeChallenge/SensorController.java

```diff
@@ -26,70 +26,56 @@
+            sensor.setSensorId(newSensorId);
             return sensorService.addNewSensor(sensor);
         }
-        else if(!sensorService.exists(sensor.getSensorId()))
-        {
+        else if(sensorService.findById(sensor.getSensorId()) == null){
             return sensorService.addNewSensor(sensor);
         }
-        else{
+        else {
             log.info("Sensor ID number taken");
             throw new IllegalStateException("Index is out of bounds");
         }
     }
 
-    public Sensor registerNewMetrics (@RequestBody Sensor sensor){
-        return sensorService.addNewSensor(sensor);
-    }
-
-    // Need to create new instance of sensorId/cityName/countryName with updated metricId/metrics/time
-    @PostMapping(path = "/{id}")
+    //POST for adding metrics values to a sensor
+    @PostMapping(path = "/{sensorId}")
     @ResponseStatus(HttpStatus.CREATED)
-    public Sensor updateSensor(@RequestBody Metrics metrics, @PathVariable String id){
-        if(!sensorService.exists(id)) {
-            log.info("Sensor ID number {} has not been registered", id);
-            throw new IllegalStateException("Index is out of bounds");
-        }
+    public Sensor updateSensor(@RequestBody Metrics metrics, @PathVariable String sensorId) {
 
-        else {
-            log.info("Sensor ID is: {}", id);
             //Find sensor to insert metrics into
-            Sensor sensor = sensorService.findById(id);
-            //Checks to see if inital sensor object is populated, if not, it fills it
-            //If metrics are empty
-            if(sensor.getMetrics() == null){
-                //Sets time
-                metrics.setLocalDateTime(LocalDateTime.now());
-                //Sets metrics
-                sensor.setMetrics(metrics);
-                //Creates metrics object with id, metrics and time
-                metricsService.createNewVersion(metrics);
-                return sensorService.addNewSensor(sensor);
-            }
-            //If metrics are full, create a new version with updated metrics
-            else{
-                metrics.setLocalDateTime(LocalDateTime.now());
-                sensor.setMetrics(metrics);
-                metricsService.createNewVersion(metrics);
-                return sensorService.addNewSensor(sensor);
-                //return sensorService.createNewVersion(sensor);
+            Sensor sensor = sensorService.findById(sensorId);
+            if (sensor == null) {
+                log.info("Sensor ID number {} has not been registered", sensorId);
+                throw new IllegalStateException("Index is out of bounds");
             }
+            else {
+                //Checks to see if initial sensor object is populated, if not, it fills it
+                if (sensor.getMetrics() == null) {
+                    sensor.setLocalDateTime(LocalDateTime.now());
+                    sensor.setMetrics(metrics);
+                    return sensorService.addNewSensor(sensor);
```

Comment:
I took a look at `addNewSensor` and it looks like it just calls `save` which is fine for updating an existing entry, but you should rename this method to something like `saveSensor` to make it clear that you're not actually adding a new sensor here but updating an existing one